Automattic/wp-calypso
# Store|Woo: Update/store redirect dashboard

#### Changes proposed in this Pull Request
This PR fixes the following in tandem with https://github.com/Automattic/wp-calypso/pull/48194:

* #48156 for products 
* #48160 for settings
* #48159 for reviews
* #48158 for promotions
* #48157 for orders


Adds redirection for store pages in the following paths to Dashboard page `/store/:site`
```
/store/products/:site
/store/product/:site
/store/product/:site/:product_id
/store/products/categories/:site
/store/products/category/:site/:category_id
/store/products/category/:site
/store/orders/:site
/store/orders/:filter/:site
/store/order/:site/:order_id
/store/order/:site/
/store/promotions/:site
/store/promotion/:site
/store/promotion/:site/:promotion_id
/store/reviews/:site
/store/reviews/:filter/:site
/store/reviews/:product_id/:filter/:site
/store/settings/:site
/store/settings/payments/:site
/store/settings/shipping/:site
/store/settings/shipping/zone/:site/:zone?
/store/settings/taxes/:site
/store/settings/email/:site/:setup?
```

#### Testing instructions

Make sure you have an eCommerce account on wordpress.com

1. Start Calypso locally
2. Navigate to http://calypso.localhost:3000/ 
3. Click "Switch Site" from the sidebar and choose your eCommerce site.
4. Directly navigate to paths mentioned above (put URL in browser and press enter), replacing the path's parameters with values. Note that only `:site` is significant to determine the redirection, other parameters will be ignored. Append the URL with `?flags=woocommerce/store-removed`.
5. Confirm that you are redirected to Store's Dashboard page.

**Testing MC stats**
1. Before starting Calypso, ensure the `mc_analytics_enabled` config in your preferred environment is set to true.
1. Follow the testing instruction above.
1. Check MC stat with the stat group `calypso_store_post_sunset`. The path you visited should be incremented.

**URL replacement example:**

http://calypso.localhost:3000/home/calypsotest363078866.wpcomstaging.com

to

http://calypso.localhost:3000/store/products/calypsotest363078866.wpcomstaging.com?flags=woocommerce/store-removed
